modules/gapi/samples/onevpl_infer_single_roi.cpp

```diff
@@ -353,29 +424,42 @@ int main(int argc, char *argv[]) {
 
     cv::GMetaArg descr = cap->descr_of();
     auto frame_descr = cv::util::get<cv::GFrameDesc>(descr);
+    auto inputs = cv::gin(cap);
 
     // Now build the graph
     cv::GFrame in;
     auto size = cv::gapi::streaming::size(in);
-    auto roi = custom::LocateROI::on(size, std::cref(device_id));
-    auto blob = cv::gapi::infer<custom::FaceDetector>(in);
-    cv::GArray<cv::Rect> rcs = cv::gapi::parseSSD(blob, size, 0.5f, true, true);
-    auto out_frame = cv::gapi::wip::draw::renderFrame(in, custom::BBoxes::on(rcs, roi));
-    auto out = cv::gapi::streaming::BGR(out_frame);
-
     cv::GStreamingCompiled pipeline;
-    try {
+    if (opt_roi.has_value()) {
```

Comment:
tried, but `pipeline` construction is also different for both cases. Since 1 only if is not enough to cover these scenarios when i would prefer to put it as is, which is similar with `infer_single_roi` sample

Reply:
thanks, done